File: terraform/src/main/kotlin/com/hyfly/tf/actuator/processor/BaseProcessor.java
package com.hyfly.tf.actuator.processor;

import lombok.Data;

@Data
public class BaseProcessor implements IProcessor {

    protected boolean hasErr = false;

    protected boolean completed = false;

    protected final StringBuilder errorBuilder;

    BaseProcessor() {
        errorBuilder = new StringBuilder();
    }

    @Override
    public void parse(String line) {
        System.out.println(line);
    }

    @Override
    public void parseError(String line) {
        System.out.println(line);
    }

    @Override
    public String getErrMsg() {
        if (hasErr) {
            return errorBuilder.toString().trim();
        }
        return null;
    }
}

File: terraform/src/main/kotlin/com/hyfly/tf/actuator/processor/BaseProcessor.kt
package com.hyfly.tf.actuator.processor

open class BaseProcessor : IProcessor {

    var hasErr: Boolean = false

    var completed: Boolean = false

    val errorBuilder: StringBuilder = StringBuilder()

    override fun parse(line: String?) {
        println(line)
    }

    override fun parseError(line: String?) {
        println(line)
    }

    override val errMsg: String?
        get() {
            return if (hasErr) {
                errorBuilder.toString().trim()
            } else {
                null
            }
        }
}
